File: Pattern24.go
// ****
// *  *
// *  *
// *  *
// ****

package main

import "fmt"

func main() {
	var n int
	fmt.Print("Enter the row number:")
	fmt.Scanf("%d",&n)
	for i:=1;i<=n;i++{
		for j:=1;j<=n-1;j++{
			if (i==1 || i==n){
				fmt.Print("*")
			}else{
				if (j==1 || j==n-1){
					fmt.Print("*")
				}else{
					fmt.Print(" ")
				}
			}
		}
		fmt.Println()
	}
}

File: Pattern24.py

# ****
# *  *
# *  *
# *  *
# ****

n = int(input("Enter the row number:"))
for i in range(1,n+1):
    for j in range(1,n):
        if (i==1 or i==n):
            print("*",end='')
        else:
            if (j==1 or j==n-1):
                print("*",end='')
            else:
                print(" ",end='')
    print()
